Software/Python/ir_remote_control/gobox_ir_receiver_libs/di_ir_reader.py

```diff
@@ -0,0 +1,486 @@
+                592,535,557,561,622,1596,
+                568,553,569,528,595,528,
+                564,1653,591,1656,559,1654,
+                570,1648,587,535,567,1650,
+                595],
+                
+                #KEY_OK 
+                [584,534,569,555,
+                567,531,591,532,570,554,
+                568,529,594,530,562,561,
+                572,1646,568,1651,594,1651,
+                563,1655,569,1655,590,1651,
+                564,1654,570,1649,596,528,
+                564,560,564,533,588,536,
+                566,556,567,530,592,1653,
+                561,536,597,1648,570,1648,
+                563,1657,588,1657,567,1652,
+                563,1660,585,535,567,1651,
+                593],
+                
+                #KEY_H 
+                [595,528,565,558,
+                564,533,589,535,567,556,
+                567,530,592,533,574,550,
+                568,1651,564,1655,589,1656,
+                569,1649,570,1647,594,1655,
+                558,1654,562,1656,589,535,
+                571,1647,593,530,563,1655,
+                590,534,568,556,566,1652,
+                562,562,561,1656,569,555,
+                567,1651,564,560,562,1656,
+                573,1646,595,529,563,1656,
+                589],
+                
+                
+                #KEY_S 
+                [614,507,560,562,
+                571,526,596,528,564,559,
+                563,534,589,535,568,556,
+                566,1652,561,1657,589,1655,
+                569,1650,569,1649,592,1653,
+                561,1657,568,1650,594,528,
+                568,1650,591,533,569,555,
+                568,529,593,530,562,1657,
+                588,536,570,1649,592,531,
+                571,1647,598,1647,567,1652,
+                563,1655,589,535,567,1651,
+                594]
+                ]
+
+    for key in range(len(key_map)):
+        total_diff=0
+        for i in range(len(inp)):
+            diff=abs(inp[i]-key_map[key][i])
+            if diff<100:
+                diff=0
+            total_diff+=diff
+            # print total_diff,
+        if total_diff < 1500:
+            found_flag=1
+            break
+                
+    # Send to socket client
+    try:
+        # Create a TCP/IP socket
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+        server_address = ('localhost', 10000)
+        # print 'connecting to %s port %s' % server_address
+        sock.connect(server_address)
+
+        # print 'sending "%s"' % str(i)
+        if found_flag:
+            print keys[key]
+            sock.sendall(keys[key])
+        else:
+            print "NO_MATCH"
+            sock.sendall("NO_MATCH")
+    except socket.error:
+        print "Unable to connect to the server"
+    finally:
+        # print 'Closing socket'
+        sock.close()
+    
+def match_with_button(inp):
+    list_to_parse=[]
+    large_val_found=0
+    if debug:
+        print inp,len(inp)
+        
+    if len(inp)==65:
```

Comment:
There is a little bit written here https://github.com/karan259/GoPiGo-1/blob/a2c2690fd47d10bd8e0ccb7e792fa5d74cb82654/Software/Python/ir_remote_control/gobox_ir_receiver_libs/di_ir_reader.py#L382-L387